examples/research_projects/codeparrot/scripts/codeparrot_training.py

```diff
@@ -256,13 +262,15 @@ def get_lr():
 model.train()
 completed_steps = 0
 t_start = time.time()
+loss_tracking = 0
 for step, batch in enumerate(train_dataloader, start=1):
     if args.resume_from_checkpoint and step < resume_step:
         continue  # we need to skip steps until we reach the resumed step
     loss = model(batch, labels=batch, use_cache=False).loss
-    log_metrics(
-        step, {"lr": get_lr(), "samples": step * samples_per_step, "steps": completed_steps, "loss/train": loss.item()}
-    )
+    avg_loss = loss.repeat(args.train_batch_size)
+    avg_loss = accelerator.gather(avg_loss).mean()
+    loss_tracking += avg_loss.item()
```

Comment:
we could already divide by `args.gradient_accumulation_steps` here so we don't need to do it later on.
```suggestion
    loss_tracking += avg_loss.item() / args.gradient_accumulation_steps
```